zcode/builder.go

```diff
@@ -31,11 +33,15 @@ func (b *Builder) BeginContainer() {
 }
 
 // EndContainer closes the most recently opened container.  It panics if the
-// receiver has no open container.
-func (b *Builder) EndContainer() {
+// receiver has no open container.  isSet must be true if and only if the
+// container is a set.
+func (b *Builder) EndContainer(isSet bool) {
```

Comment:
I think adding this boolean makes for a worse interface, but I'm not sure what else to do. I'd have preferred `EndContainer(typ *zng.Type)`, but that causes a circular import dependency.

Reply:
This is just cosmetic, but how about renaming `isSet` to `normalize` ?  It seems like the builder API is lower-level and doesn't know about zng types.

Along those lines, and related to your comment about the interface, another option would be to leave the `BeginContainer()` and `EndContainer()` unchanged, and having a separate `NormalizeContainer()` call that a client can call any time while in a container (with the common and expected case being to call it just before calling `EndContainer()`. Just an idea, entirely your call.

Reply:
How about this option:
- keep EndContainer() as is before the PR.
- add an additional call, `EndContainerTransform(func transform(zcode.Bytes) zcode.Bytes))`, that does the same as EndContainer, but calls the passed in `transform` with the container bytes, and replaces them with the return result?

Reply:
I went with a variation on @alfred-landrum's suggestion that I found more ergonomic:
* Keep `func (b *Builder) EndContainer()`
* Add `func (b *Builder) TransformContainer(transform func(Bytes) Bytes)`